orka_vector_api/helper/gdal_helper.py

```diff
@@ -1,29 +1,81 @@
-# TODO make this actually work
-def create_or_append_geopackage(bbox, data_id, conn, app):
-    host = app.config['PG_HOST'],
-    port = app.config['PG_PORT'],
-    database = app.config['PG_DATABASE'],
-    user = app.config['PG_USER'],
-    password = app.config['PG_PASSWORD']
-
-    filename = data_id + '.gpkg'
-
-    # TODO add proper filepath
-    # TODO use SQL methods of psycopg to provide proper sql query here
-    sql = ''
-    # TODO get layername
-    layername = ''
+import os
+import subprocess
+
+import yaml
+from psycopg2.sql import SQL, Identifier, Literal
+
+
+def _get_geopackage_cmd(filename, layername, sql, host=None, port=None, database=None, user=None, password=None):
     cmd = f'ogr2ogr -f "GPKG" {filename} ' \
           f'PG:"host={host} user={user} port={port} dbname={database} password={password}" ' \
           f'-sql "{sql}" ' \
           f'-nln "{layername}" ' \
           f'-append'
 
-    cur = conn.cursor()
-    cur.execute(sql)
-    result = cur.fetchall()
-    for geom in result:
-        print(geom)
+    return cmd
+
+
+def create_geopackage(data_id, app, layer_sqls):
+    gpkg_path = app.config['ORKA_GPKG_PATH']
+    filename = os.path.abspath(os.path.join(gpkg_path, data_id + '.gpkg'))
+    db_props = {
+        'host': app.config['PG_HOST'],
+        'port': app.config['PG_PORT'],
+        'database': app.config['PG_DATABASE'],
+        'user': app.config['PG_USER'],
+        'password': app.config['PG_PASSWORD']
+    }
+
+    for layername, layer_sql in layer_sqls:
+        cmd = _get_geopackage_cmd(filename, layername, layer_sql, **db_props)
+        proc = subprocess.run(cmd, shell=True, capture_output=True)
+        app.logger.info(f'Exporting {layername} exited with code {proc.returncode}')
+        if proc.stderr:
+            app.logger.error(f'[stderr]\n{proc.stderr.decode()}')
+        if proc.returncode != 0:
+            return False
+    return True
```

Comment:
Some additional loggers (e.g. printing out the `gdal` command) may be useful here.